src/game/GameHud.cpp

```diff
@@ -0,0 +1,466 @@
+
+    float HP = p.getMarine()->getCurrentHealth();
+
+    if (HP > MIN_HEALTH) {
+        setHealthBarColor(HP); //Sets the RGB values of the healthbar from the marine's current HP
+        healthBarForeground.w = static_cast<size_t>(HP / MAX_HEALTH * healthBarBackground.w) -
+                                healthBarBackground.h * HEALTHBAR_FOREGROUND_W_RAT;
+
+        //Sets the renderers color based on calculated RGB value
+        SDL_SetRenderDrawColor(Renderer::instance().getRenderer(), getHealthRgbElement(0),
+                                getHealthRgbElement(1), getHealthRgbElement(2), OPAQUE);
+
+        //Renders the healthbar foreground
+        SDL_RenderFillRect(Renderer::instance().getRenderer(), &healthBarForeground);
+    }
+}
+
+/**
+ * Function: renderConsumable
+ *
+ * Date:
+ * JF: April 1, 2017:
+ *
+ * Designer:
+ * Jacob Frank
+ *
+ * Programmer:
+ * Jacob Frank
+ *
+ * Interface: renderConsumable(SDL_Rect screenRect, Player p)
+ *                  SDL_Rect screenRect: The Current screen properties (height, width)
+ *                  Player p: The player in the game. used to find current equipped weapon
+ *
+ * Notes:
+ * Function, when called renders the consumable item slot in the bottom left corner of the visible screen
+ * Consumable slot is only visible when the player has a consumable item in their inventory.
+ */
+void GameHud::renderConsumable(const SDL_Rect screenRect, const Player p) {
+
+    if (screenRect.w <= screenRect.h) {
+        consumableSlot.w =  screenRect.w * CONSUMABLE_SIZE_RAT;
+        consumableSlot.h =  screenRect.w * CONSUMABLE_SIZE_RAT;
+    } else {
+        consumableSlot.w = screenRect.h * CONSUMABLE_SIZE_RAT;
+        consumableSlot.h = screenRect.h * CONSUMABLE_SIZE_RAT;
+    }
+
+    consumableSlot.x = screenRect.w * PADDING_RAT;
+    consumableSlot.y = screenRect.h - screenRect.w * PADDING_RAT - consumableSlot.h;
+
+    Renderer::instance().render(consumableSlot, TEXTURES::CONSUMABLE_SLOT);
+    Renderer::instance().render(consumableSlot, TEXTURES::HEALTHPACK);
+
+}
+
+/**
+ * Function: renderWeaponSlots
+ *
+ * Date:
+ * JF: April 1, 2017:
+ *
+ * Designer:
+ * Jacob Frank
+ *
+ * Programmer:
+ * Jacob Frank
+ *
+ * Interface: renderWeaponSlots(SDL_Rect screenRect, Player p)
+ *                  SDL_Rect screenRect: The Current screen properties (height, width)
+ *                  Player p: The player in the game. used to find current equipped weapon
+ *
+ * Notes:
+ * Function, when called renders the Weapon inventory slots along the bottom center of the visible screen
+ * Inventory slots are only visible when the player changes their currently equiped weapon.
+ */
+void GameHud::renderWeaponSlots(const SDL_Rect screenRect, const Player p) {
+    int weaponSlotWidth = screenRect.w * WEAPON_SLOT_WIDTH_RAT;
+    int weaponSlotHeight = screenRect.h * WEAPON_SLOT_HEIGHT_RAT;
+    int weaponSlotPosY = screenRect.h - screenRect.w * PADDING_RAT - weaponSlotHeight;
```

Comment:
You give me a feeling of consternation.
I'm constantly concerned about your variable declarations
Could you possibly construct a new thing?
Is this being subtle?
These need const